### ISSUE TITLE
`index-out-of-bounds` ignores `>` tuple length checks

### ISSUE THREAD

**Issue by andyli-oai** [astral-sh/ty#4451]
### Summary

Greater-than length check does not narrow a union of one-element and two-element tuples, resulting in an `index-out-of-bounds` diagnostic on valid access.

### Reproduction

```python
def foo(values: tuple[int] | tuple[int, int]) -> int | None:
    if len(values) > 1:
        return values[1]
    return None
```
[Playground](https://play.ty.dev/c6768d0b-023d-4a4c-8496-12f82246551c)

### Error

```text
error[index-out-of-bounds]: Index 1 is out of bounds for tuple `tuple[int]` with length 1
```

### Expected

No diagnostic.

### Comparison with other conditions

Replacing the condition gives these results:

- `len(values) >= 2`: same diagnostic.
- `len(values) == 2`: no diagnostic.
- `len(values) != 1`: no diagnostic.

### Related
An [existing comment on closed #560](https://github.com/astral-sh/ty/issues/560#issuecomment-4096877313) also describes an analogous problem with a string-length `>` guard.

### Version

0.0.77
